firestore/query.go

```diff
@@ -103,7 +119,12 @@ func (q Query) Where(path, op string, value interface{}) Query {
 		q.err = err
 		return q
 	}
-	q.filters = append(append([]filter(nil), q.filters...), filter{fp, op, value})
+	proto, err := filter{fp, op, value}.toProto()
```

Comment:
Should this be factored out as a function?

Reply:
Great catch. I can use WherePath to replace all that follow.s